schemas/parameters/schema/parameter.schema.json

```
@@ -65,16 +70,39 @@
     "parameter_type",
     "default_value"
   ],
-  "if": {
-    "properties": {
-      "parameter_type": {
-        "const": "state"
+  "allOf": [
+    {
+      "if": {
+        "properties": {
+          "default_value": {
+            "not": {
+              "const": null
+            }
+          }
+        }
+      },
+      "then": {
+        "not": {
+          "required": [
+            "optional"
+          ]
+        }
+      }
```

I'm not certain about the logical implications here; does "not: required" equate to "forbidden"? if you've tested it with VSCode parser and it does what you expect, then that's great. But in my head it makes more sense if you phrase it the other way around: don't include the `optional` property to begin with, but if default value is null, then include `optional`and make it required.

```suggestion
      "if": {
        "properties": {
          "default_value": {
            "const": null
          }
        }
      },
      "then": {
        "properties": {
          "optional": {
            "description": "Specify if this parameter if optional if the default value is null",
            "type": "boolean"
          },
          "required": [
            "optional"
          ]
        }
      }
```

---

As explained in person, I believe that `not: required` actually means forbidden, especially with the VSCode parser.